Docs/source/Software/Tasks.rst

```diff
@@ -42,3 +42,82 @@ Process:
     2) If all of these checks are safe, the task will send the All Clear message and the Contactor On message across the CAN line.
     
     3) The task will then destroy itself since it is no longer needed
+
+CLI Task: Sugam Arora
+=====================
+
+Purpose
+    The Command Line Interface prints metrics and information about the BPS depending on what command the user has entered, and is meant to be used to debug the BPS.
+
+Functionality
+    This task initializes the CLI and polls for a command to be sent via UART (through USB). Once the user has entered a command, it is handled and the appropriate information is displayed.
+
+Priority
+    The CLI task has priority level 9 because it is the least important task (besides the idle task).
+
+Shared Resources
+    This task may use the ``MinionsASIC_Mutex`` or the ``Voltage_Mutex`` depending on what command the user enters.
+
+Timing Requirements
+    None
+
+Yields
+    This task will wait until the user enters a command. This task will yield for the resources mentioned in the "Shared Resources" section for this task.
+
+Additional Considerations
+    For information on how to use the CLI and its list of valid commands, click on the CLI section in "Application" on this site.
+
+
+Battery Balancing Task: Sugam Arora
+===================================
+
+Purpose
+    The solar car is powered by lithium ion cells. If these cells are overcharged, they may experience thermal runaway. Running the battery with unbalanced cells may lead to its accelerated degredation. This task has been 
+    implemented in an effort to maintain peak performance and health for the car's battery pack.
```

Comment:
Also include that other modules will stop charging if one cell contain a high voltage, leading to an overall lower state of charge.